matscipy/dislocation.py

```diff
@@ -2509,193 +2549,19 @@ def build_impurity_cylinder(self, disloc, impurity, radius,
-
-        burgers = alat * np.array([1, -2, 1.]) / 6.
-
-        disloCenterX = 0.5 * (alat * np.linalg.norm(axes[0])) / 6.0
-        # 1/4 + 1/2 * (1/3 - 1/4) - to be in the middle of the glide set
-        disloCenterY = 7.0 * (alat * np.linalg.norm(axes[1])) / 24.0
-
-        unit_cell_core_position = np.array([disloCenterX,
-                                            disloCenterY, 0])
-
-        parity = [0, 0]
-
-        unit_cell = Diamond(symbol, directions=axes.tolist(),
-                            pbc=(False, False, True),
-                            latticeconstant=alat)
-
-        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
-
-        n_planes = 2
-        # There is very small distance between
-        # atomic planes in glide configuration.
-        # Due to significant anisotropy application of the self consistent
-        # displacement field leads to deformation of the atomic planes.
-        # This leads to the cut plane crossing one of the atomic planes and
-        # thus breaking the stacking fault.
-        self_consistent = False
-        super().__init__(unit_cell, alat, C11, C12, C44,
-                         axes, burgers, unit_cell_core_position, parity,
-                         glide_distance, n_planes=n_planes,
-                         self_consistent=self_consistent)
-
-
-class DiamondGlide90degreePartial(CubicCrystalDislocation):
-    def __init__(self, alat, C11, C12, C44, symbol='C'):
-        axes = np.array([[1, 1, -2],
-                         [1, 1, 1],
-                         [1, -1, 0]])
-
-        burgers = alat * np.array([1., 1., -2.]) / 6.
-
-        disloCenterX = 0.5 * (alat * np.linalg.norm(axes[0])) / 6.0
-        # 1/4 + 1/2 * (1/3 - 1/4) - to be in the middle of the glide set
-        disloCenterY = 7.0 * (alat * np.linalg.norm(axes[1])) / 24.0
-
-        unit_cell_core_position = np.array([disloCenterX,
-                                            disloCenterY, 0])
-
-        parity = [0, 0]
-
-        unit_cell = Diamond(symbol, directions=axes.tolist(),
-                            pbc=(False, False, True),
-                            latticeconstant=alat)
-
-        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
-
-        n_planes = 2
-        # There is very small distance between
-        # atomic planes in glide configuration.
-        # Due to significant anisotropy application of the self consistent
-        # displacement field leads to deformation of the atomic planes.
-        # This leads to the cut plane crossing one of the atomic planes and
-        # thus breaking the stacking fault.
-        self_consistent = False
-        super().__init__(unit_cell, alat, C11, C12, C44,
-                         axes, burgers, unit_cell_core_position, parity,
-                         glide_distance, n_planes=n_planes,
-                         self_consistent=self_consistent)
-
-
 class CubicCrystalDissociatedDislocation(CubicCrystalDislocation):
```

Comment:
This could be an abstract base class to prevent it being accidentally instatiated (same for `CubicCrystalDislocation`). I think this is simply `from abc import ABC` then subclass `ABC`.

Reply:
Adding metaclass=ABCMeta was the solution here, as making CubicCrystalDissociatedDislocation directly subclass ABC created issues in determining the method resolution order (MRO; see https://stackoverflow.com/questions/2010692/what-does-mro-do)